google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleStage.java

```diff
@@ -55,17 +58,21 @@ public AppEngineFlexibleStage(
   public void stage(@NotNull Path stagingDirectory) {
     try {
       Path stagedArtifactPath = stagingDirectory.resolve(
-          "target" + AppEngineFlexDeploymentArtifactType.typeForPath(deploymentArtifactPath));
+          "target" + AppEngineFlexibleDeploymentArtifactType.typeForPath(deploymentArtifactPath));
       Files.copy(deploymentArtifactPath, stagedArtifactPath);
 
+      // Uses user-specified app.yaml and Dockerfile files if they exist. Auto generates them
+      // otherwise.
       Path appYamlPath = deploymentConfiguration.isAuto()
+          || !Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))
           ? helper.defaultAppYaml()
           : Paths.get(deploymentConfiguration.getAppYamlPath());
       Files.copy(appYamlPath, stagingDirectory.resolve("app.yaml"));
 
       Path dockerFilePath = deploymentConfiguration.isAuto()
+          || !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))
```

Comment:
hmm. if the configuration is set to "Custom" but the dockerfile path does not exist, I would expect an error (if its runtime: custome), not usage of the automatically generated one